src/core/public/application/capabilities/capabilities_service.tsx

```diff
@@ -38,7 +38,9 @@ export interface CapabilitiesStart {
  */
 export class CapabilitiesService {
   public async start({ appIds, http }: StartDeps): Promise<CapabilitiesStart> {
+    const useDefaultCapabilities = http.anonymousPaths.isAnonymous(window.location.pathname);
     const capabilities = await http.post<Capabilities>('/api/core/capabilities', {
+      query: useDefaultCapabilities ? { useDefaultCapabilities } : undefined,
       body: JSON.stringify({ applications: appIds }),
     });
```

Comment:
Just to be 100% clear: what overlap does an anonymous path have with `route.options.authRequired`?

At first when reviewing this it seemed like the check in the Spaces capability switcher would be redundant because of this, but after digging a bit, I'm really not sure.

Reply:
Same question here.

From the issue's description

> For optionally authenticated routes, the existing system poses a challenge: it's not possible for all switchers to know if the request should be authenticated or not

But this changes does not really impacts optionally authenticated routes, does it?

Reply:
Sorry for the confusion here!

> At first when reviewing this it seemed like the check in the Spaces capability switcher would be redundant because of this, but after digging a bit, I'm really not sure.

It may be redundant, but I view it as a belt and suspenders approach. It's possible to request capabilities server-side without going through this client-side `capabilities_service`, so I aired on the side of resiliency here. 

>  what overlap does an anonymous path have with route.options.authRequired
> But this changes does not really impacts optionally authenticated routes, does it?

You're right, it rather impacts unauthenticated & optionally authenticated _pages_. The fact that a route is optionally authenticated isn't as relevant here. I shouldn't have conflated the two.